DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi/FrontEnd/src/client/app/activity/activity.service.js

```diff
@@ -0,0 +1,40 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('app.activity')
+        .factory('ActivityService', activityService);
+
+    activityService.$inject = ['$q', '$http', 'ACTIVITY_CONSTANTS'];
+
+    function activityService($q, $http, constants) {
```

Comment:
what I proposed you was to have here something like an array with already loaded external activities from backend. so that if you are just clicking to pages 1-2-1-2-1-2 each new click is not generating new request. service can be good place for some kind of cache.

but this approach can have some drawbacks. (like it is not clear when to invalidate this cache)